plugins/include/open62541/plugin/nodesetloader.h

```diff
@@ -0,0 +1,27 @@
+/* This work is licensed under a Creative Commons CCZero 1.0 Universal License.
+ * See http://creativecommons.org/publicdomain/zero/1.0/ for more information.
+ *
+ *    Copyright 2019 (c) Julius Pfrommer, Fraunhofer IOSB
+ */
+
+#ifndef UA_NODESET_LOADER_DEFAULT_H_
+#define UA_NODESET_LOADER_DEFAULT_H_
+
+#include <open62541/util.h>
+
+_UA_BEGIN_DECLS
+
+UA_EXPORT void
+UA_NodesetLoader_Init(void);
```

Comment:
Dear @jpfr,

Regarding the **_ini**t method, I put the method concept-wise, since I needed a **_delete** method for the custom data types. I will delete it. Also for the naming, I will resolve that as well.

But for the **_delete** method, I checked and as I'm aware of the server does not have an option to automatically handle the deletion of custom data types. Therefore, the user must manually cleanup memory for custom data types. I saw the use-case within the following example: [server_types_custom.c:304](https://github.com/open62541/open62541/blob/737f0562fcaabe4906f985981754e850e97b0c72/examples/custom_datatype/server_types_custom.c#L304).

Could you please specify if
>Having a Boolean "tag" in the DataTypeArray to indicate that they should be automatically cleaned up together with the server
>
is something that is existing or is a task to be implemented?

Best regards,
Srdjan Usorac

Reply:
The structure is defined here:
https://github.com/open62541/open62541/blob/b457206d248b8ddd139f2eb29016af87e86f4377/include/open62541/types.h#L1071

If you add the flag here (for example as this)

```c
typedef struct UA_DataTypeArray {
    const struct UA_DataTypeArray *next;
    const size_t typesSize;
    const UA_DataType *types;
    UA_Boolean cleanup; /* Free the array structure and its content
                                  when the client or server configuration
                                  containing it is cleaned up */
} UA_DataTypeArray;
```

then you can release the memory more cleanly as part of the ClientConfig and ServerConfig cleanup.
Because the array will be inserted into these configs.

With that you don't need the _init and _delete logic.
So the NodeSetLoader API becomes simples as suggested above.

Reply:
Nice, thank you for the suggestions.